From 549dd475693bb0cddb7b9dd41377f1c125ba876e Mon Sep 17 00:00:00 2001
From: Rolando Roca <rolandor@valvesoftware.com>
Date: Fri, 22 Sep 2023 14:23:12 -0700
Subject: [PATCH 015/104] Bluetooth: Fixed an issue where the Steam Deck could
 not into sleep because it continued advertizing after the Steam Controller
 was disconnected.

<Change Type>: Poor experience

<<< Test Notes >>>
<Test-Proposal>:
1)BT BR/EDR gamepad wakeup function
1)BT BLE gamepad wakeup function
<Stress-Test>: Y

[Fwd-ported from v6.5]
Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 net/bluetooth/hci_sync.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/bluetooth/hci_sync.c b/net/bluetooth/hci_sync.c
index cbc3a75d7..589aba62a 100644
--- a/net/bluetooth/hci_sync.c
+++ b/net/bluetooth/hci_sync.c
@@ -6218,7 +6218,7 @@ int hci_suspend_sync(struct hci_dev *hdev)
 
 	if (hci_conn_count(hdev)) {
 		/* Soft disconnect everything (power off) */
-		err = hci_disconnect_all_sync(hdev, HCI_ERROR_REMOTE_POWER_OFF);
+		err = hci_disconnect_all_sync(hdev, HCI_ERROR_REMOTE_USER_TERM);
 		if (err) {
 			/* Set state to BT_RUNNING so resume doesn't notify */
 			hdev->suspend_state = BT_RUNNING;
-- 
2.53.0

